### PR TITLE
Use different procedure names with different versions of GDS

### PR DESCRIPTION
**Pull request by ali-ince**
This PR adds a way to use different procedure names on different versions of GDS and also adds integration tests for latest versions of 4.4 and 5 enterprise neo4j versions.

### CODE REVIEW

File: .mypy.ini [hunk 1]
```diff
@@ -1,4 +1,4 @@
 [mypy]
-python_version = 3.8
```
Comment (fbiville) [inline .mypy.ini:2]: is this change required?

File: poetry.lock [hunk 2]
```diff
@@ -20,18 +31,18 @@ tests-no-zope = ["cloudpickle", "hypothesis", "mypy (>=1.1.1)", "pympler", "pyte
 
 [[package]]
 name = "bandit"
-version = "1.7.5"
+version = "1.7.6"
 description = "Security oriented static analyser for python code."
 optional = false
-python-versions = ">=3.7"
+python-versions = ">=3.8"
```
Comment (fbiville) [inline poetry.lock:37]: same question here

**Review by fbiville — APPROVED** [review]

File: .mypy.ini (hunk 1 above)
Comment (ali-ince) [reply .mypy.ini:2]: we already had specified python versions >= 3.9 in pyproject.toml and also in our tox tests - so wanted to make sure that we also check for language compatibility from that version onward.

File: poetry.lock (hunk 2 above)
Comment (ali-ince) [reply poetry.lock:37]: I don't have any control over this, which seems to be coming from this dependency's new version.
